src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -461,6 +461,11 @@ void Connection::websocket_connected_handler(const std::string& protocol)
                     (value_2 >= get_oldest_supported_protocol_version() && value_2 <= get_current_protocol_version());
                 if (good_version) {
                     logger.detail("Negotiated protocol version: %1", value_2);
+                    // Grab the connection ID from the websocket if the protocol version is less than 10,
+                    // otherwise, this value will come in a server log message
+                    if (value_2 < 10) {
+                        receive_appservices_coid(m_websocket->get_appservices_request_id());
+                    }
```

Comment:
hm, can we just always do this for now? otherwise this would block @yaseenmongo's work on the server change to start sending these, and that has not been prioritized yet.

Reply:
Do you mean the protocol update to v10 has been delayed?
This is here to retain the previous functionality until the server protocol is updated to v10.

Reply:
> Or is the co_id not going to be sent until later, even though the new bind format and server log messages will be supported sooner?

right - the new bind format is being supported first which will involve the protocol version bump on the server (blocker for the breaking changes project). support for the log messages will come later (but that should does not need to block merging this client-side support for it)